docs/cli.md

```diff
@@ -124,6 +124,14 @@ prettier --single-quote --list-different "src/**/*.js"
 
 You can also use [`--check`](cli.md#--check) flag, which works the same way as `--list-different`, but also prints a human-friendly summary message to stdout.
 
+## `--diff`
```

Comment:
I prefer `--diff` only work when doing `--check`

```bash
prettier --check --diff .
```


WDYT?

Reply:
My only thought against it would be that if anyone wanted to pipe the output to a diff file that could be applied later by a diff tool, requiring `--check` would mean that they have to strip the summary (unless you mean adding `--diff` would implicitly remove the summary).

But I don't really feel strongly about it and am fine either way.

Reply:
> My only thought against it would be that if anyone wanted to pipe the output to a diff file that could be applied later by a diff tool

Make sense.

Reply:
How about:


For files

```js
prettier . --diff
```

Format and show diff


```js
prettier . --write --diff
```

Not allow.


```js
prettier . --check --diff
```

Format and show diff

---

For stdin

```js
echo "code" | prettier . --diff
```

Format and show diff


```js
echo "code" | prettier . --write --diff
```

Not allow.


```js
echo "code" | prettier . --check --diff
```

Format and show diff.

-----

So `--diff` always do the same as `--check`, but show different message, and it can use together with `--check`, but not `--write`.